test/src/org/apache/jmeter/protocol/http/sampler/TestHTTPSamplers.java

```diff
@@ -364,4 +369,23 @@ public void testSetAndGetMimetype() {
         file = sampler.getHTTPFiles()[0];
         assertEquals("mime2", file.getMimeType());
     }
+
+    @Test
+    public void testRawBodyFromFile() {
+        String baseDirPath = FileServer.getFileServer().getBaseDir();
+        File baseDir = new File(baseDirPath);
+        try {
+            FileServer.getFileServer().setBase(baseDir.getParentFile());
+            HTTPSamplerBase sampler = new HTTPSampler3();
+            sampler.setMethod("POST");
+            sampler.setPath("http://httpbin.org/post");
+            sampler.setHTTPFiles(new HTTPFileArg[]{new HTTPFileArg("bin/jmeter.properties", "", "")});
```

Comment:
Why, why do you people use `bin/jmeter.properties` for unit testing?
Eh :(

Reply:
We have introduced a `resources` directory under `test` that should be used for test files -- and those should be test specific, but it was either really old code, or someone was lazy.